SpringBox/pair_interactions.py

```diff
@@ -2,57 +2,75 @@
 from scipy.spatial.distance import pdist, squareform
 
+
 def active_particles(pXs, prv_acc, activation_fn, _config):
     # Particles that were previously active, now can loose their activation with a rate of activation_decay_rate
     prv_acc = prv_acc * (np.random.rand(len(prv_acc)) > _config['activation_decay_rate'] * _config['dt'])
     # Or particles can become active when they enter the light activated area
     new_acc = activation_fn(pXs)
     return np.maximum(prv_acc, new_acc)
 
+
 @numba.jit
-def spring_forces(acc,pXs , Dij, _config):
+def spring_forces(acc, pXs, Dij, Drxns, _config):
     rhs = np.zeros_like(pXs)
-    n_part=_config['n_part']
-    k=_config['spring_k']
-    r0=_config['spring_r0']
-    Iij = Dij * np.outer(acc,acc)
-    Iij = (Iij>_config['spring_lower_cutoff']) * (Iij<_config['spring_cutoff'])
+    n_part = _config['n_part']
+    k = _config['spring_k']
+    r0 = _config['spring_r0']
+    Iij = Dij * np.outer(acc, acc)
+    Iij = (Iij > _config['spring_lower_cutoff']) * (Iij < _config['spring_cutoff'])
     for i in range(n_part):
-        for j in range(i+1,n_part):
-            if Iij[i,j]!=0:
-                rhs[i] += -k*((Dij[i,j]-r0)/Dij[i,j])*(pXs[i]-pXs[j])
-                rhs[j] += +k*((Dij[i,j]-r0)/Dij[i,j])*(pXs[i]-pXs[j])
+        for j in range(i + 1, n_part):
+            if Iij[i, j] != 0:
+                rhs[i] += -k * ((Dij[i, j] - r0) / Dij[i, j]) * (Drxns[i][j])
+                rhs[j] += +k * ((Dij[i, j] - r0) / Dij[i, j]) * (Drxns[i][j])
     return rhs
+
+
 @numba.jit
-def LJ_forces(acc,pXs , Dij, _config):
-    n_part=_config['n_part']
+def LJ_forces(acc, pXs, Dij, _config):
+    n_part = _config['n_part']
     rhs = np.zeros_like(pXs)
-    eps=_config['LJ_eps']
-    r0=_config['LJ_r0']
-    r0_6=r0**6
-    LJ_pre = 12*eps*r0_6
-    Dij6 = Dij**6
-    np.clip(Dij6, 1e-6, None, out = Dij6)
-    Iij = (Dij<_config['LJ_cutoff']) * np.outer(acc,acc)
+    eps = _config['LJ_eps']
+    r0 = _config['LJ_r0']
+    r0_6 = r0 ** 6
+    LJ_pre = 12 * eps * r0_6
+    Dij6 = Dij ** 6
+    np.clip(Dij6, 1e-6, None, out=Dij6)
+    Iij = (Dij < _config['LJ_cutoff']) * np.outer(acc, acc)
     for i in range(n_part):
-        for j in range(i+1,n_part):
-            if Iij[i,j]!=0:
-                rhs[i] += -LJ_pre*(pXs[i]-pXs[j])*((Dij6[i,j]-r0_6)/(Dij[i,j]*Dij6[i,j]))
-                rhs[j] += +LJ_pre*(pXs[i]-pXs[j])*((Dij6[i,j]-r0_6)/(Dij[i,j]*Dij6[i,j]))
+        for j in range(i + 1, n_part):
+            if Iij[i, j] != 0:
+                rhs[i] += -LJ_pre * (pXs[i] - pXs[j]) * ((Dij6[i, j] - r0_6) / (Dij[i, j] * Dij6[i, j]))
+                rhs[j] += +LJ_pre * (pXs[i] - pXs[j]) * ((Dij6[i, j] - r0_6) / (Dij[i, j] * Dij6[i, j]))
     return rhs
 
+
+@numba.jit
+def periodic_dist(u, v, length):
+    x_diff = min(abs(u[0] - v[0]), 2 * length - abs(u[0] - v[0]))
+    y_diff = min(abs(u[1] - v[1]), 2 * length - abs(u[1] - v[1]))
+    return np.sqrt(x_diff ** 2 + y_diff ** 2)
+
+
+def vectors(pXs, L):
```

Comment:
Two comments here: 
- Vectors is a bad name, since it does not describe what the function does, and it possibly clutters the namespace with functions that have a definitive meaning (shadowing other functions) Can you name this something like periodic_difference_vectors?
- Either here or at the place where you call it, you need to check if we want to have periodic boundary conditions or not. We only want to compute periodic difference vectors if we run our simulation with periodic boundary conditions.